File: CS/02_Text/InsertSimpleHTMLString/InsertSimpleHTMLString.cs
using Spire.Pdf;
using Spire.Pdf.Graphics;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace InsertSimpleHTMLString
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            //Create a pdf document
            PdfDocument doc = new PdfDocument();

            //Add a new page
            PdfNewPage page = doc.Pages.Add() as PdfNewPage;

            //HTML string
            string htmlText = "This demo shows how we can insert <u><i>HTML styled text</i></u> to PDF using "
                             + "<font color='#FF4500'>Spire.PDF for .NET</font>. ";

            //Render HTML text
            PdfFont font = new PdfFont(PdfFontFamily.Helvetica, 25);
            PdfBrush brush = PdfBrushes.Black;
            PdfHTMLTextElement richTextElement = new PdfHTMLTextElement(htmlText, font, brush);
            richTextElement.TextAlign = TextAlign.Left;

            //Format Layout
            PdfMetafileLayoutFormat format = new PdfMetafileLayoutFormat();
            format.Layout = PdfLayoutType.Paginate;
            format.Break = PdfLayoutBreakType.FitPage;

            //Draw htmlString  
            richTextElement.Draw(page, new RectangleF(0, 20, page.GetClientSize().Width, page.GetClientSize().Height), format);
   

            String result = "InsertSimpleHTMLString-result.pdf";
            doc.SaveToFile(result);

            //Launch the Pdf file
            PDFDocumentViewer(result);
        }
        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }                    
     
        }
    }
}

File: VB/02_Text/InsertSimpleHTMLString.vb
Imports Spire.Pdf
Imports Spire.Pdf.Graphics

Namespace InsertSimpleHTMLString
    Partial Public Class Form1
        Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            ' Create a new PdfDocument
            Dim doc As New PdfDocument()

            ' Add a new page to the document
            Dim page As PdfNewPage = TryCast(doc.Pages.Add(), PdfNewPage)

            ' Specify the HTML string to be inserted
            Dim htmlText As String = "This demo shows how we can insert <u><i>HTML styled text</i></u> to PDF using " & "<font color='#FF4500'>Spire.PDF for .NET</font>. "

            ' Create a PdfFont using the Helvetica font and size 25
            Dim font As New PdfFont(PdfFontFamily.Helvetica, 25)

            ' Create a PdfBrush to set the text color
            Dim brush As PdfBrush = PdfBrushes.Black

            ' Create a PdfHTMLTextElement instance to load and render the HTML text
            Dim richTextElement As New PdfHTMLTextElement(htmlText, font, brush)
            richTextElement.TextAlign = TextAlign.Left

            ' Define the layout format for the HTML text
            Dim format As New PdfMetafileLayoutFormat()
            format.Layout = PdfLayoutType.Paginate
            format.Break = PdfLayoutBreakType.FitPage

            ' Draw the HTML text on the page within the specified rectangle and format
            richTextElement.Draw(page, New RectangleF(0, 20, page.GetClientSize().Width, page.GetClientSize().Height), format)

            ' Save the document to a file
            Dim result As String = "InsertSimpleHTMLString-result.pdf"
            doc.SaveToFile(result)

            ' Close the document
            doc.Close()

            'Launch the Pdf file
            PDFDocumentViewer(result)
        End Sub
        Private Sub PDFDocumentViewer(ByVal fileName As String)
            Try
                Process.Start(fileName)
            Catch
            End Try

        End Sub
    End Class
End Namespace
